das/store.go

```diff
@@ -0,0 +1,112 @@
+package das
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+	"golang.org/x/net/context"
+)
+
+var (
+	storePrefix   = datastore.NewKey("das")
+	checkpointKey = datastore.NewKey("checkpoint")
+)
+
+// The checkpoint store stores/loads the DASer's checkpoint to/from
+// disk using the checkpointKey. The checkpoint is stored as a struct
+// representation of the latest successfully DASed state.
+type store struct {
```

Comment:
I am against creating an abstraction over datastore when you can just pass in a wrapped namespace to begin with, and use `datastore.Datastore` directly

Reply:
I think its fine in this case

Reply:
@Wondertan When I question an abstraction, it almost never has to do with performance.  It purely has to do with simplicity of the code - the abstraction here doesn't do anything helpful when it comes to understanding the code. The separation of concerns you mentioned was an argument I had about merging two structs that were abstractions over separate things - it does not apply here

Reply:
https://github.com/celestiaorg/celestia-node/issues/1049